kernel/mempool.c

```diff
@@ -98,7 +93,7 @@ void k_thread_system_pool_assign(struct k_thread *thread)
 #define _SYSTEM_HEAP	NULL
 #endif
 
-void *z_thread_malloc(size_t size)
+void *z_thread_aligned_malloc(size_t align, size_t size)
```

Comment:
It might be a bit more consistent to name it aligned_alloc rather than aligned_malloc, just for consistency.